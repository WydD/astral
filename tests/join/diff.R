key,value
t=1
A,27
B,28
C,29
t=2
A,27
B,31
t=3
A,28
C,31
t=4
D,15

