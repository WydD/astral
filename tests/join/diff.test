<?xml version="1.0" encoding="UTF-8"?>
<astral xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://astral.ligforge.imag.fr/schema/query.xsd">
	<query>
		<sigma condition="value != oldvalue">
			<join defaultOuter="0" outer="full">
				<source id="R"/>
				<domain dtype="previous">
					<rho>
						<rename from="value" to="oldvalue"/>
						<source id="R"/>
					</rho>
				</domain>
			</join>
		</sigma>
	</query>
	<sources>
		<source file="tests/join/diff.R" id="R" impl="FileRelationImpl"/>
	</sources>
</astral>
