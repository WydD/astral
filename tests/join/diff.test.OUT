key,value,oldvalue
t=1
A,27,0
B,28,0
C,29,0
t=2
C,0,29
B,31,28
t=3
B,0,31
A,28,27
C,31,0
t=4
A,0,28
C,0,31
D,15,0

